### DISCUSSION TITLE
Composable returned refs are not reactive inside pinia store

### DISCUSSION THREAD

**Discussion by saver711** [vuejs/pinia#2836] [Help and Questions] [answered]
### Reproduction

https://play.pinia.vuejs.org/#eNqNVEtu2zAQvQqrTRzEEhOkSIHACdwWWaSLtmiy1EaWxjYTiRRIyhFgeN1lL9ID9Dy9QK/QR+pj2U3cGLBlcWbevHnDmXXwviyjVUXBZTAxqRalZYZsVbI8kYurOLAmDq5jKYpSacvWrDJ0Z5UmtmFzrQp2FPFUVdKSjqw52nW8p6Ic+OEktDgaej0lNl32PuABYyxTJQ14+DxXfcrR8da2ZnWNMG90aRobw8cjjup6zFZJXiH+mq0bC2MuVuUU5WoxOkqXqFDIhUMSEjZwkiTtURsJRBezwXPCG2kgBF5cDXliCW+MTWaVtUqyaZqL9BGC1fXJCSS7lammAnAOft3Q3Ux4491EvgnDZxF83ZHcgwFGa9gHCkM8JnxAKxgHjcJhkZTRg1ES7fUqxK0BXb3sdImDUkiRuJM44P5/RKYIZ1o9GfT1Ac7jznWKHvGMVlap3IRJKZqwpbWlueQ8zST8M8rFSkeSLJdlwf+JmV5EF9EZz8WMIxEXMqN6Nw0iwoyK16B3rtPT6Oxt9M6jrs5b0MKhOtBNLDeQxRrcgblY7Inimi9y0l9KK3BHdsRJ8lw9ffJnVlfUU0yXlD4+c/5g6ob1V02Qb0WDsmyiF2Qb883dZ6rxvzcWKqtyeB8wfiPc38pxbNw+VDID7YGfZ3vre4y7fW9uakvSdEU5ol4N7+9l/nig9C3d8+h8oOJ24iFhP8sZzYXc2w7+Ng0HXtN8/OzUv25xxJJq79esgX4bXQ2zY7Ybghjl0XG7Af63OV7eG4e2hh/JnY3h9oVH5JzdL3EDWAJ+UlmUjvEUK2JWMbskJoxBnt8/v//59cNFNvwkqEGk0WnLS2MbawnamHuoj1M0oFNkrwMviasBhW8r3bySqev1VgFUCnPHAJX1FNrIF+TxlgMCeQoC1y9D0ajY71iTzPKBZB4DZfWZuoI7dORsndzD/wSbv8ndSGA=

### Steps to reproduce the bug

Click on increment xx
and observe the browser console

### Expected behavior

changing xx from inside the composable 1
changing xx in component 1
changing xx in store 1

### Actual behavior

changing xx from inside the composable 1
changing xx in component 1

### Additional information

Iam kinda new to Vue, if iam missing something here let me know plz
and what to do if i need the updated x value inside the store ?

**Comment by posva** [comment, marked as answer]
A composable allows to reuse logic across components (and stores in this case). You have two different `xx` variables updating. Create a store instead of a composable if you want to share them

**Reply by saver711** [reply]
I got it
the state is changing inside the composable and in the place that is changing it, right ?

**Reply by posva** [reply]
Each time you call the composables, a new xx is created. With a store, no matter the amount of calls, it’s only one
